### DISCUSSION TITLE
Flow Update Data Operation creates invalid SQL when updating multiple rows

### DISCUSSION THREAD

**Discussion by sgreiner** [directus/directus#17507] [Archive]
### Describe the Bug

We want to create a flow to update some data.
From a "Run Script" Operation we get an Array of "id"s.
We input this Array via template notation as input for the "Update Data" in the "IDs" field of the operation.
No data is updated if there is more than one id in the array, instead we get an error in the Flow-Log.

After looking through https://github.com/directus/directus/blob/main/api/src/operations/item-update/index.ts I think the operation is expected to support more than one id.

The problem is the SQL which gets generated:
```SQL
`id` in ((1, 2, 3))
```
there is one pair of parentheses to much. If I manually remove one pair, then I can execute that SQL.
Our current workaround is to create a comma separated string of ids and put them in as query filter.

### To Reproduce

Create a new flow.

Add Operation:

Run Script:
```JavaScript
module.exports = async function(data) {
    return {
        keys_to_update: [1,2,3]
    };
}
```

Add Operation:

Update Data with {{$last.keys_to_update}} used as IDs
![image](https://user-images.githubusercontent.com/2290942/218513453-0d62449b-2e73-4998-add6-cfbc204159e8.png)

### Errors Shown

Error-Result:
```JSON
{
  "sql": "update `components` set `field` = 'new value', `user_updated` = 'xxx', `date_updated` = '2023-02-13 14:21:39.609' where `id` in ((1, 2, 3))",
  "code": "ER_OPERAND_COLUMNS",
  "errno": 1241,
  "index": 0,
  "sqlState": "21000",
  "sqlMessage": "Operand should contain 1 column(s)"
}
```

### What version of Directus are you using?

9.22.4

### What version of Node.js are you using?

bundled with the official Docker Image 

### What database are you using?

MySQL 5.7.39

### What browser are you using?

Chrome

### How are you deploying Directus?

Docker

**Comment by connorwinston** [comment]
The ID's field is an array; by adding it and clicking enter, you add that template notation as an item to the array. You should use the curly brackets to the right of "IDs" and just input the template notation.
<img width="834" alt="image" src="https://user-images.githubusercontent.com/12714889/218645413-890a09ae-4033-4dc8-9b7d-222d429d0e96.png">

**Reply by sgreiner** [reply]
Thank you, that helped 😃

**Comment by u12206050** [comment]
I have encountered the same problem, and although it worked the first time I removed the `{{$trigger.body.keys}}` from the array, when I edited it again, it saved it again within an array.

So there seems to be two parts to fix to this. In general `whereIn` shouldn't result in a double `(())` being generated, and on the frontend if using curly braces then the array should be parsed correctly if that is the intention.
